bundles/http/Citrix/SAP-Concur-Demo-HTTP/script.js

```diff
@@ -0,0 +1,603 @@
+  console.log('Synching data...')
+
+  countTotalAmount()
+
+  dataStore.save('reports', reports)
+  dataStore.save('expenses', expenses)
+  dataStore.save('categories', categories)
+  console.log("Synchronization complete")
+}
+
+function countTotalAmount() {
+  let amounts = new Map()
+  expenses.forEach(e => {
+    amounts.set(e.ReportName, (amounts.has(e.ReportName)? amounts.get(e.ReportName) : 0) + e.Amount)
+  })
+  reports.forEach(r => r.TotalAmount = amounts.get(r.Id))
+}
+
+async function incrementalSynch({dataStore, client}) {
+  console.log("Running incremental synchronization...")
+  //pulling a random report from the report list
+  let report = JSON.parse(JSON.stringify(reports[Math.floor(Math.random() * (reports.length))]))
+  //this looks really scary, but we need to get the string value of report id, transfer it to int, add the randomly generated value...
+  const reportId = parseInt(report.Id) + Math.floor(Math.random() * (1000 - reports.length)) + reports.length
+  //and then return it back to being a string
+  let rid = reportId.toString()
+  report.Id = rid
+  report.DateSubmitted=date.toLocaleDateString()
+
+  //randomly generating a number of expenses for a report
+  let numOfExpenses = Math.floor(Math.random() * (7 - 1)) + 1
+  //expenses will be saved into the array 
+  let iExpenses = []
+  let totalamount = 0
+  //pulling the last id of an expense so we can adequately increment the ids
+  //we have to slice the value since the ids are far outside of integer range 
+  let eLastId = parseInt(expenses[expenses.length-1].Id.slice(12,17))
+  for (let i=0; i< numOfExpenses; i++) {
+    //this monstrosity is required if we want to make a deep copy of an object
+    //otherwise we get a shallow copy and any change to the object also changes the expense in the global variable
+    //which leads to SQL conflicts further down the line
+    let expense = JSON.parse(JSON.stringify(expenses[Math.floor(Math.random() * (expenses.length))]))
+    expense.ReportName = rid
+    expense.TransactionDate=report.DateSubmitted
+    let eId = eLastId + (i + 1)
+    expense.Id = expense.Id.slice(0,12) + eId.toString()
+    totalamount += expense.Amount
+    if (i==numOfExpenses-1) expenses.push(expense)
+    iExpenses.push(expense)
+  }
+  report.TotalAmount=totalamount
+  dataStore.save('reports', report)
+  dataStore.save('expenses', iExpenses)
+  console.log("Incremental synchronization completed...")
+}
+
+async function updateReport({ dataStore, client, actionParameters, integrationParameters }) {
+  console.log(`Updating report with id ${actionParameters.Id}...`)
+  
+  const {Id, Type, OwnerName, OwnerMail, DateSubmitted, TotalAmount, CurrencyCode, Status} = actionParameters
+  const report = {Id, Type, OwnerName, OwnerMail, DateSubmitted, TotalAmount, CurrencyCode, Status}
+
+  dataStore.save('reports', report)
+  console.log(`Report with id ${actionParameters.Id} updated`)
+}
+
+integration.define({
```

Comment:
undocumented convention - better to start the integration script with this declaration, i.e.  this should be on the first line